ttnn/cpp/ttnn/operations/eltwise/binary_ng/device/binary_ng_utils.hpp

```diff
@@ -75,7 +75,8 @@ struct OpConfig {
 void add_activation_defines(
     std::map<std::string, std::string>& defines,
     tt::stl::Span<const unary::UnaryWithParam> activations,
-    std::string_view operand);
+    std::string_view operand,
+    const DataType& dtype = DataType::BFLOAT16);
```

Comment:
Please do not default this. If you want a default, make this `std::optional<DataType> dtype = std::nullopt` for this and `unary::utils::get_op_init_and_func`, and `TT_FATAL(dtype.has_value(), ...);` in `unary::utils::get_op_init_and_func` where it is required to be set before accessing with `*dtype`.